openfasoc/generators/gdsfactory-gen/process_input.py

```diff
@@ -0,0 +1,233 @@
+			self.print_to_stream("Hello!")
+			self.print_to_stream("Please provide a name for the Component you want to create")
+			self.print_to_stream("remember, this will be the name of your top level component: ")
+			self.name = self.read_from_stream().strip()
+		else:
+			self.name = str(toplvlname).strip()
+		# save a list of responses to recreate this component from a .conv file
+		self.conversation_responses = list()
+		self.conversation_responses.append(self.name)
+		# init the code object
+		self.code = GlayoutCode(self.name)
+		# create a backup that goes back exactly one call to process_next_input
+		self.backup = self.__backup()
+	
+
+	def __backup(self):
+		newobj = self.__class__
+		backup = newobj.__new__(newobj)
+		backup.inputstream = self.inputstream
+		backup.outputstream = self.outputstream
+		backup.conversation_responses = copy.deepcopy(self.conversation_responses)
+		backup.code = copy.deepcopy(self.code)
+		backup.backup = None
+		return backup
+	
+	def save_to_disk(self, savecode: bool=True, savepath: Optional[Union[str,Path]]=None):
+		"""Save NLP results to disk, either save code or conversation responses.
+		Args:
+			savemode: bool, default to True==save code, false==save conversation responses
+			savepath: Optional (defaults to None), 
+			NOTE: None is the same as specifying the current directory "./"
+		"""
+		# figure out which save mode we are in
+		suffix = ".py" if savecode else ".convo"
+		if savepath is None:
+			savepath = Path("./").resolve()
+		savepath = Path(savepath.strip()) if isinstance(savepath,str) else savepath
+		if savepath.is_dir():
+			savepath /= f"{self.code.toplvl_name}{suffix}"
+		if savepath.parent.is_dir():
+			with savepath.open("w") as savefile:
+				if savecode:
+					savefile.write(self.code.get_code())
+				else:
+					savefile.writelines([r+"\n" for r in self.conversation_responses])
+		else:
+			raise FileNotFoundError("the directory you specfied does not exist")
+	
+	def print_to_stream(self, toprint: Optional[str]=None):
+		if toprint is None:
+			toprint=""
+		self.outputstream.write(toprint+"\n")
+	
+
+	def read_from_stream(self) -> str:
+		response = self.inputstream.readline()
+		#self.conversation_responses.append(response)
+		return response
+	
+	def __save_response(self, response: str):
+		self.conversation_responses.append(response.strip())
+
+	def process_next_input(self, text_input: str) -> bool:
+		"""main driver for doing things
+		returns True if session is ongoing
+		"""
+		self.backup = self.__backup()
+		sentences = nltk.sent_tokenize(text_input)
+		for sentence in sentences:
+			saveresponse = True
+			sentence = sentence.strip().removesuffix(".")
+			words = nltk.word_tokenize(sentence)
+			mode_indicator = words[0].strip().replace("-","").lower()
+			# parse user input
+			if mode_indicator[0]=="h":# help
+				saveresponse = False
+				syntaxes = list()
+				syntaxes.append("place an genidhere called/named compnamehere with paramshere")#place_syntax
```

Comment:
It would be good to create a separate class method for printing the help documentation and use it here.

Reply:
done